AddonTerraformTemplate/variables.tf

```diff
@@ -37,3 +37,8 @@ variable "resource_group_location" {
   default     = "eastus"
   description = "Location of the resource group."
 }
+
+variable "useAzureMonitorPrivateLinkScope" {
```

Comment:
Why is this only being done for terraform? Can we include bicep, arm, policy too in this?

Reply:
Thanks Kaveesh! I am doing this change for terraform currently to mitigate the issue. I will do the other template changes after hearing back from PMs on the scenarios.

Reply:
I'm not sure I fully understand, what is the question that you're waiting to hear back from the PMs? Don't the other onboarding scenarios also need to be mitigated/have the same scenario which is onboarding a private AKS cluster?

We should keep the templates consistent across onboarding mechanisms.